pkg/replication/replicator.go

```diff
@@ -375,25 +392,32 @@ func (txr *TxReplicator) fetchNextTx() error {
 	}
 
 	if syncReplicationEnabled {
-		md := exportTxStream.Trailer()
+		bMayCommitUpToTxID, ok := emd["may-commit-up-to-txid-bin"]
```

Comment:
This metadata returned can be a type and you can add a validate method to do this validation

Reply:
metadata is a generic placeholder, in the context of this PR it's limited to implement bidirectional streaming. Having a dynamic checking might be beneficial in the future as the metadata is used extensively, but I'd suggest to keep it simpler in the scope of this PR

Reply:
I don't think adding a small function to validate that would make it complex, even if we keep it as a map, because it is a lot of validation happening atm